### DISCUSSION TITLE
The difference between swarm learning and decentralized federated learning

### DISCUSSION THREAD

**Discussion by xiby** [HewlettPackard/swarm-learning#2] [General]
Hello, 
    I am studying decentralized federated learning(DFL), and read your artical on _nature_. However, I found that DFL and swarm learning(SL) are so alike, such as they both train local model without leakage of private data, and they do not need a server to aggregate the global model. So I want to know what's the essential difference between SL and DFL.
    Looking forward to your reply, thanks.

**Comment by sathya-SL** [comment]
Please refer papers or articles that talk about DFL for adding context to the discussion

**Comment by xiby** [comment]
I will provide some papers I have read: 

1. [Fully Decentralized Federated Learning](http://bayesiandeeplearning.org/2018/papers/140.pdf)
2. [Peer-to-peer Federated Learning on Graphs](https://arxiv.org/abs/1901.11173)
3. [Blockchain Assisted Decentralized Federated Learning (BLADE-FL): Performance Analysis and Resource Allocation](https://arxiv.org/abs/2101.06905)
4. [BAFFLE : Blockchain Based Aggregator Free Federated Learning](https://arxiv.org/abs/1909.07452)
5. [D-Cliques: Compensating NonIIDness in Decentralized Federated Learning with Topology](https://arxiv.org/abs/2104.07365)
6. [A Blockchain-Based Decentralized Federated Learning Framework with Committee Consensus](https://ieeexplore.ieee.org/abstract/document/9293091)
